### PR TITLE
builder/picolib: add needed file for compiling math functions with error support

### PR DESCRIPTION
**Pull request by deadprogram**
This PR updates builder/picolib to add a needed file for compiling math functions with error support.

Thanks to @aykevl for actually finding and providing this fix, I really just reported the problem and tested that the fix worked as expected.

### CODE REVIEW

File: builder/picolibc.go [hunk 1]
```diff
@@ -337,6 +337,7 @@ var picolibcSourcesLarge = []string{
 	"libm/common/math_err_may_uflow.c",
 	"libm/common/math_err_check_uflow.c",
 	"libm/common/math_err_check_oflow.c",
+	"libm/common/math_errf_invalidf.c",
```
Comment (aykevl) [inline builder/picolibc.go:341]: We might want to add all the other `math_err_*.c` files here too.

**Review by aykevl — COMMENTED** [review]
It would be nice to have a test for this fix. Can you add a variant of the testcase to testdata/math.go? (basically `float32(math.Sqrt(float64(someValue)))`).

File: builder/picolibc.go (hunk 1 above)
Comment (deadprogram) [reply builder/picolibc.go:341]: OK doing it now.

Comment (deadprogram) [reply builder/picolibc.go:341]: Done.

**Comment by deadprogram** [comment]
Thanks for the help resolving this @aykevl now merging.

**Pull request merged**